### PR TITLE
use "true" and "false" instead of 0 and 1

### PR DESCRIPTION
**Pull request by samos123**
Jax/XLA seem to expect true and false strings instead of using 0 and 1.

### CODE REVIEW

**Review by ruomingp — APPROVED** [review]
Thanks! Is there a test that can added or updated?

**Review by markblee — APPROVED** [review]
+1 on adding test case.

File: axlearn/common/compiler_options_test.py [hunk 1]
```diff
@@ -24,10 +24,10 @@ def f(x: Tensor) -> Tensor:
         )
         self.assertEqual(f_compiled(5), 15)
 
-    def atest_xla_flags_from_options(self):
```
Comment (samos123) [inline axlearn/common/compiler_options_test.py:27]: this test was never being run because of the `a` before `test` in function name.

**Comment by samos123** [comment]
I updated the existing test and renamed the test so it's being run correctly now.

File: axlearn/common/compiler_options.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def xla_flags_from_options(xla_options: dict[str, Union[str, bool, int]]) -> str
     flags = []
     for k, v in xla_options.items():
         if isinstance(v, bool):
-            v = "1" if v else "0"
+            v = "true" if v else "false"
```
Comment (apghml) [inline axlearn/common/compiler_options.py:147]: Should we convert all the "true" and "false" strings in `default_xla_options` to bools to be consistent?

Comment (apghml) [inline axlearn/common/compiler_options.py:147]: Would it make sense to do an aot compilation or run the trainer for a few steps to confirm these changes work? (If you haven't already?)

File: axlearn/common/compiler_options.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def xla_flags_from_options(xla_options: dict[str, Union[str, bool, int]]) -> str
...
         flags.append(f"--{k}={v}")
```
Comment (apghml) [inline axlearn/common/compiler_options.py:148]: Could you clarify whether there are two types of flags, bool and tristate, or just one type, tristate?

Comment (samos123) [reply axlearn/common/compiler_options.py:148]: there are 2 types of flags, but a bool should be converted to "true" or "false" so it can be used for either type of flag.

File: axlearn/common/compiler_options.py (hunk 2 above)
Comment (samos123) [reply axlearn/common/compiler_options.py:147]: I did test it on v6e afair last year, but any additional runs would be great.

Comment (samos123) [reply axlearn/common/compiler_options.py:147]: I think that should be preferred. I just wanted to keep this PR small to make merging and reviewing easier. Happy to include that change in this PR or do a follow up.

Comment (apghml) [reply axlearn/common/compiler_options.py:147]: Maybe test it again on v5p using the current jax/jaxlib version pinned in AXLearn?

Comment (apghml) [reply axlearn/common/compiler_options.py:147]: Thanks, I think it makes sense to include in this PR.

Comment (markblee) [reply axlearn/common/compiler_options.py:147]: Were you able to run this?

**Review by markblee — APPROVED** [review]

File: axlearn/common/compiler_options.py (hunk 2 above)
Comment (samos123) [reply axlearn/common/compiler_options.py:147]: No I don't have readily access to all TPUs. I only have occasional access. Is there no CI that runs tests on PRs so we can verify changes like this during pre-submit?

**Comment by samos123** [comment]
It's still relevant. @muyangyuapple encountered the same issue on pathways. But feel free to discard this and make a separate PR.

tl;dr we should always true or false instead of 1 or 0 for a flag is boolean like.
